fastlane_bot/helpers/txhelpers.py

```diff
@@ -813,21 +818,22 @@ def _get_layer_one_gas_fee_loop(self, rawTransaction) -> Decimal:
             self.ConfigObj.GAS_ORACLE_CONTRACT.caller.l1FeeScalar()
         ))
 
-        return self._get_layer_one_gas_fee(rawTransaction, ethereum_base_fee, fixed_overhead, dynamic_overhead)
-
-    def _get_layer_one_gas_fee(self, rawTransaction, ethereum_base_fee: int, fixed_overhead: int, dynamic_overhead: int) -> Decimal:
-        """
-        Returns the expected layer one gas fee for a layer 2 Optimism transaction
-        :param rawTransaction: the raw transaction
-        :param ethereum_base_fee: the L1 base fee received from the contract
-        :param fixed_overhead: the fixed overhead received from the contract
-        :param dynamic_overhead: the dynamic fee received from the contract
-        returns: Decimal
-            The total fee (in gas token) for the l1 gas fee
-        """
         zero_bytes, non_zero_bytes = count_bytes(rawTransaction)
         tx_data_gas = zero_bytes * 4 + non_zero_bytes * 16
         tx_total_gas = (tx_data_gas + fixed_overhead) * dynamic_overhead
         l1_data_fee = tx_total_gas * ethereum_base_fee
         ## Dividing by 10 ** 24 because dynamic_overhead is returned in PPM format, and to convert this from WEI format to decimal format (10 ** 18).
         return Decimal(f"{l1_data_fee}e-24")
+
+    def _get_layer_one_gas_fee_2(self, rawTransaction) -> Decimal:
+        """
+        Returns the expected layer one gas fee for a Mantle (layer two) transaction
+        :param rawTransaction: the raw transaction
+
+        returns: Decimal
+            The total fee (in gas token) for the l1 gas fee
+        """
+
+        l1_data_fee = asyncio.get_event_loop().run_until_complete(asyncio.gather(self.ConfigObj.GAS_ORACLE_CONTRACT.caller.getL1Fee(rawTransaction)))
+        ## Dividing by 10 ** 18 in order to convert from wei resolution to native-token resolution
+        return Decimal(f"{l1_data_fee}e-18")
```

Comment:
This change can be reverted, as it will be obtained directly from a similar task already completed for Base and Optimism (which would otherwise lead to manual conflict when this change is merged).